src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -743,7 +743,7 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
                     library => Path.GetFullPath(Path.Combine(
                         Path.GetDirectoryName(assetsFile.PackageSpec.RestoreMetadata.ProjectPath),
                         PathUtility.GetPathWithDirectorySeparator(library.MSBuildProject))),
-                    library => new PackageIdentity(library.Name, library.Version));
+                    library => new PackageIdentity(library.Name, library.Version), PathUtility.GetStringComparerBasedOnOS());
```

Comment:
Can you please put the Comparer in a different line. 

It was a bit tricky to read without an editor.

Reply:
@NuGet/nuget-client fyi.